lib/ansible/module_utils/ansible_tower.py

```diff
@@ -83,6 +93,30 @@ def tower_check_mode(module):
             module.fail_json(changed=False, msg='Failed check mode: {0}'.format(excinfo))
 
 
+def sanitise_and_merge_variables(*args):
+    '''tower_auth_config attempts  to parse variables in json/yaml format including from a file.
+    If multiple arguments are provided they are merged left to right with the right having priority.
+    Returns a json encoded string
+    '''
+    result = {}
+    for var in args:
+        if isinstance(var, string_types):
+            if var.startswith('@'):
+                filename = os.path.expanduser(var[1:])
+                with open(filename, 'r') as f:
```

Comment:
This should really be called from the controller in an action_plugin if the expectation is the input file is aligned with the playbook.  In addition, it should use the `loader` instance from the action_plugin and not call `open` directly.